### DISCUSSION TITLE
How to restart vaultwarden automatically?

### DISCUSSION THREAD

**Discussion by gittbuh** [dani-garcia/vaultwarden#2020] [Q&A]
I am running vaultwarden inside a proxmox container. When the container restarts, vaultwarden does not. I have to do it manually it seems. Is there a good reason for this? How can I make it start automatically?

Thanks!

**Comment by RealOrangeOne** [comment]
This isn't necessarily about having vaultwarden _restart_ automatically, just about having it _start_ automatically.

Doing it by default isn't a very commonly done thing. Let the user do it when they want to. How to do it is generally very easy, it just depends on how you've deployed it?

**Reply by gittbuh** [reply]
I see thanks. I just cannot see why it would not be a very commonly done thing. Is it for security reasons? At first sight, it appears to me as rather inconvinient, I am just wondering what the advantages are. 

I installed the docker image following the documentation.

**Reply by RealOrangeOne** [reply]
It is a very commonly done thing, it's just that in most examples it's left out, because quite often people forget its there and don't necessarily want to start it. Lots of linux services don't automatically enable themselves with systemd on installation for exactly the same reason: Safe defaults.

Regarding getting this running, you'll want to set `restart: unless-stopped` in your compose file. That way it'll restart on boot, unless you stop it manually (which is normally what you want). There's some more information on how automatic starting works here: https://docs.docker.com/config/containers/start-containers-automatically/
